python/pyspark/ml/tuning.py

```diff
@@ -500,13 +501,13 @@ class CrossValidatorModel(Model, _CrossValidatorParams, MLReadable, MLWritable):
     .. versionadded:: 1.4.0
     """
 
-    def __init__(self, bestModel, avgMetrics=[], subModels=None):
+    def __init__(self, bestModel, avgMetrics: List = None, subModels=None):
```

Comment:
Out of curiosity ‒why do we need default here?

Reply:
Good question, I don't think these should be optional

Reply:
At the same time, this would change the public API